contracts/contracts/Umbra.sol

```diff
@@ -9,8 +9,18 @@ import "@openzeppelin/contracts-ethereum-package/contracts/token/ERC20/SafeERC20
 contract Umbra is Ownable, ReentrancyGuard {
     using SafeMath for uint256;
 
-    event EthAnnouncement(address indexed receiver, uint256 indexed amount, string note);
-    event TokenAnnouncement(address indexed receiver, uint256 indexed amount, address indexed token, string note);
+    event Announcement(
+        address indexed receiver,
+        uint256 indexed amount,
+        address indexed token,
+        bytes16 iv,  // Inivitalization Vector
+        bytes32 pk0, // Ephemeral Public Key
```

Comment:
Small thing—instead of `pk0` and `pk1` we should use `pkx` and `pky`, just so it's more clear which is the x-coordinate of the public key and which is the y-coordinate.

Aside from this, looks and works great!

Reply:
Good call, done!